src/urlschemehandler.cpp

```diff
@@ -216,6 +218,33 @@ UrlSchemeHandler::replyZimNotFoundPage(QWebEngineUrlRequestJob *request,
     request->reply("text/html", buffer);
 }
 
+void
+UrlSchemeHandler::replyBadZimFilePage(QWebEngineUrlRequestJob *request,
+                                  const QString &zimId)
+{
+    QBuffer *buffer = new QBuffer;
+    const auto& book = KiwixApp::instance()->getLibrary()->getBookById(zimId);
+    const QString path = QString::fromStdString(book.getPath());
+    const QString name = QString::fromStdString(book.getName());
+    const QString zimEntryPath = request->requestUrl().path();
+
+    QString contentHtml = "<section><div>"
+                          "<h1>" + gt("bad-zim-file-error-page-title") + "</h1>"
+                          "<p>"  + gt("bad-zim-file-error-page-text") + "</p>"
+                          "<p>"  + gt("zim-id")   + ": <b>" + zimId + "</b></p>"
+                          "<p>"  + gt("zim-name") + ": <b>" + name  + "</b></p>"
+                          "<p>"  + gt("zim-path") + ": <b>" + path  + "</b></p>"
+                          "<p>" +  gt("zim-entry-path") + ": <b>" + zimEntryPath + "</b></p>"
+                          "</div></section>";
+
+    buffer->open(QIODevice::WriteOnly);
+    buffer->write(contentHtml.toStdString().c_str());
+    buffer->close();
+
+    connect(request, SIGNAL(destroyed()), buffer, SLOT(deleteLater()));
+    request->reply("text/html", buffer);
```

Comment:
I wonder if we should not set a error code to something else than 200.
Request may be made by js script which will check the error code.
But I haven't found an API allowing to return a specific content and set error code in the same time.

Reply:
I had the same difficulty while reviewing #1120. But since there we didn't have such a potential problem with accessing a resource via js, I [didn't mind](https://github.com/kiwix/kiwix-desktop/pull/1120#issuecomment-2151808886) about that scenario. Should we use [`QWebEngineUrlRequestJob::fail()`](https://doc.qt.io/qt-5/qwebengineurlrequestjob.html#fail) in this case (but then no special information about the error will be shown to the user)?